<div class="alert alert-block alert-info">
  <h3>⌨️ Keyboard shortcut</h3>

  These common shortcuts could save your time going through this notebook:
  <ul>
    <li>Run the current cell: <b><code>Shift</code> + <code>Enter</code></b>.</li>
    <li>Add a cell above the current cell: Press <b><code>A</code></b>.</li>
    <li>Add a cell below the current cell: Press <b><code>B</code></b>.</li>
    <li>Change a code cell to markdown cell: Select the cell, and then press <b><code>M</code></b>.</li>
    <li>Delete a cell: Press <b><code>D</code></b> twice.</li>
    <li>Run an LLM prompt cell: First run <b><code>%load_ext sdo_llm_magic</code></b> (once per session), then start a cell with <b><code>%%sdo_llm</code></b> and run it with <b><code>Shift</code> + <code>Enter</code></b>.</li>
  </ul>

  Need more help with keyboard shortcut? Press <b><code>H</code></b> to look it up.
</div>

# 👋 Welcome

<div style="
background-color: rgba(0, 123, 255, 0.08);
border-left: 6px solid #4a6fa5;
padding: 16px;
border-radius: 6px;
margin: 12px 0;
">

#### 🧩 **What is proteomic data?**
Proteomic data refers to information about the proteins expressed in a cell, tissue, or organism at a given time. It captures not only which proteins are present but also their abundance, modifications, interactions, and functional states. It provides a direct view of cellular function, complementing genomic and transcriptomic information.

#### 🧠 **Why do we need proteomic data?**
In personalized medicine, applications of quantifying proteomics are typically based on biomarkers, which are measurable indicators that reflect disease risk or disease status. For example, studies show the proteomic profile of prostate cancer can be significantly altered during the course of the disease. Proteomic data is essential because proteins are the primary functional molecules in cells, executing the instructions encoded in DNA and regulated by RNA.

#### 🩺 **What types of proteomic data are used in research?**
Proteomic research generates multiple complementary data types to capture protein abundance, modifications, interactions, and spatial context.

</div>

# Algorithmic Thinking and Proteomics Data

**Module:** Algorithmic Thinking and Proteomics Data

**Paper:** Jiang *et al.* (2020), Cell — A Quantitative Proteome Map of the Human Body.

**Data subset (local):**
- `protein_expression_mx.csv` — protein × sample expression matrix
- `protein_selected_names.csv` — curated set of tissue-enriched proteins


Run every `%%sdo_llm` cell to let the LLM generate content — no coding required.


<div style="
background-color: #fff4cc;
border-left: 6px solid #f0ad4e;
padding: 16px;
border-radius: 6px;
margin-top: 10px;
font-size: 16.5px;
line-height: 1.6;
">

<b>🔎 Big picture (plain language)</b><br>

This notebook follows Jiang <i>et al.</i> (Cell, 2020) and uses GTEx-based proteomics to build a “map” of which proteins are most abundant in different human tissues. The core idea is simple: measure thousands of proteins across many tissues, then ask which proteins are consistently higher in one tissue (tissue-enriched) and what that tells us about tissue function.

In this module, we’ll inspect a protein-by-sample expression matrix, run basic quality checks and normalization, and reproduce a small slice of the paper’s message: different tissues have distinct protein signatures that can be summarized and communicated with simple, interpretable plots.

</div>


<details>
<summary style="font-size: 18px; font-weight: 700;">
🔑 Key terms you’ll see throughout this notebook (recommended)
</summary>

<div style="
background-color: #eef4ff;
border-left: 6px solid #4a6fa5;
padding: 16px;
border-radius: 8px;
margin-top: 10px;
font-size: 16.5px;
line-height: 1.6;
">

<h3>🧬 Proteomics basics</h3>
<ul>
  <li><b>Proteome</b>: The full set of proteins present in a cell/tissue.</li>
  <li><b>Proteomics</b>: Measuring many proteins at once (often by mass spectrometry).</li>
  <li><b>Protein abundance</b>: How much of a protein is present in a sample (a quantitative level).</li>
  <li><b>Tissue-enriched protein</b>: A protein that is consistently higher in one tissue than others.</li>
  <li><b>GTEx</b>: A large public resource that collected many human tissues from donors for molecular profiling.</li>
  <li><b>Donor / sample / tissue</b>: A donor is a person; a sample is a tissue piece; tissue is the organ type (e.g., liver).</li>
</ul>

<h3>📦 Dataset structure</h3>
<ul>
  <li><b>Expression matrix</b>: A table where rows are proteins and columns are samples (or tissues).</li>
  <li><b>Curated list</b>: A hand-picked subset (here, proteins already known to be tissue-enriched).</li>
  <li><b>ID alignment</b>: Making sure identifiers match across files (same protein names/IDs, same samples).</li>
</ul>

<h3>🧹 Quality control (QC)</h3>
<ul>
  <li><b>Missing values</b>: Proteins not measured (or filtered) in some samples.</li>
  <li><b>Batch effects</b>: Non-biological differences caused by how/when data were generated.</li>
  <li><b>Distribution anomalies</b>: Unusual value patterns that can indicate errors or strong outliers.</li>
</ul>

<h3>⚖️ Normalization and comparability</h3>
<ul>
  <li><b>Log2 transform</b>: A standard rescaling that compresses large values and stabilizes variation.</li>
  <li><b>Median centering</b>: Shifting samples so their medians line up (reduces technical differences).</li>
  <li><b>Quantile normalization</b>: Forces samples to have the same overall distribution (stronger adjustment).</li>
  <li><b>Robust normalization (e.g., RobNorm)</b>: Methods designed to handle outliers and heterogeneous tissues.</li>
</ul>

<h3>📊 Scoring, selection, and stability</h3>
<ul>
  <li><b>z-score</b>: “How many standard deviations above/below average” a value is.</li>
  <li><b>Thresholding</b>: Picking items that pass a cutoff (e.g., z-score &gt; a chosen value).</li>
  <li><b>Resampling / subsampling</b>: Repeating analysis on random subsets to test robustness.</li>
  <li><b>Bootstrap</b>: A common resampling method used to estimate stability/uncertainty.</li>
</ul>

<h3>🖼️ Visualization terms</h3>
<ul>
  <li><b>Bar plot</b>: Compares a small number of proteins/metrices clearly.</li>
  <li><b>Heatmap</b>: A colored grid summarizing patterns across many proteins and tissues.</li>
  <li><b>Baseline vs deviation profile</b>: Compare a “typical” profile (baseline) to tissue-specific differences.</li>
</ul>

</div>
</details>


<div style="
background-color: #fff4cc;
border-left: 6px solid #f0ad4e;
padding: 16px;
border-radius: 8px;
margin: 12px 0;
font-size: 16.5px;
line-height: 1.6;
">

<b>⚠️ Note on LLM outputs</b><br>

LLM responses are inherently non-deterministic, so repeated runs may produce slightly different wording, ordering, or intermediate steps.  
To improve reliability, keep prompts concise and explicit, and rerun a cell if the output is incomplete.

This notebook’s <code>%%sdo_llm</code> helper includes basic retry and formatting safeguards, but occasional failures can still occur.

</div>

### ⚠️ Setup: Load the sdo_llm_magic Extension (Run First)
Make sure to run the cell below before using any %%sdo_llm commands.

🔵 ▶ Run the cell below to continue.

In [5]:
%load_ext sdo_llm_magic

from pretty_llm_output import (
    install_sdo_pretty_patch,
    display_llm,
    capture_pretty,
)

install_sdo_pretty_patch()

The sdo_llm_magic extension is already loaded. To reload it, use:
  %reload_ext sdo_llm_magic


True

## Stage 0 — Ethical and Responsible AI Checklist

**In a nutshell:** Sanitize the kitchen and confirm every ingredient is legal before you touch a single pan.

**What it is:** A required checkpoint completed before any data handling or LLM prompting.

**Why it matters:** Robust data science starts with lawful, bias-aware, reproducible foundations. Skipping this stage contaminates every result that follows.

**How you do it:** Confirm consent and secondary use, governance, de-identification, HIPAA/GDPR alignment. Select a privacy model, log all prompts and model versions, and commit to source verification for any claim.

### Checklist

- Confirm consent and secondary use scope
- Verify de-identification and re-identification risk assessment
- Validate HIPAA/GDPR alignment and data governance approvals
- Select privacy model and secure compute environment
- Enable prompt, model version, and decision logging
- Commit to hallucination control and source verification

This checklist focuses on:

- **Legitimate data use:** consent, governance, de-identification, re-identification risk, HIPAA/GDPR compliance.
- **Health equity:** check sampling bias and representation across subgroups (e.g., tissue sources, demographics if available).
- **Secure compute:** encrypted, access-controlled cloud; contracts that prohibit model training on participant data.
- **Hallucination control and auditability:** cite primary sources and log every step.

### LLM prompt templates (Stage 0)

Use these prompts *before* you ask the model to inspect or analyze the GTEx proteomics data.

1. **Privacy and security risks**  
   > Identify privacy or security risks in analyzing this GTEx-based proteomics dataset. Suggest specific mitigations for data storage, access control, and model usage.

2. **Source-backed claims**  
   > For every factual claim you give about proteomics methods or biology, cite at least three peer-reviewed or authoritative sources. If no source exists, clearly mark the statement as speculative.

3. **Speculation and confidence**  
   > Review your previous answer and flag any part that is speculative, low-confidence, or based on weak evidence. Explain why and suggest how I could verify or falsify those points.


🔵 ▶ Run the cell below once and wait (up to 5 minutes) ⏳

In [6]:
%%sdo_llm --show-reply

Analyze the privacy and data-security risks of this PPGR study's data — CGM time series,
stool metagenomics/microbiome, and clinical/metabolic phenotyping on 55 deeply phenotyped
participants — and propose concrete, HIPAA-aligned mitigations. Cover, in prose:

1. Sensitive information present: direct identifiers, quasi-identifiers (age, sex,
   race/ethnicity, BMI, blood pressure, prediabetes/T2D status), PHI, and any IRB,
   data-use, or journal licensing restrictions.

2. Why re-identification risk is elevated: small cohort (n=55) plus highly individuating
   CGM traces and microbiome profiles, near-unique to a person even after de-identification.

3. Whether reproducing Figure 1c (aggregate per-meal mean glucose AUC-above-baseline and
   time-to-peak, not individual traces) lowers that risk, and how to minimize residual risk
   (aggregation thresholds, small-group suppression, what may/may not be sent to an LLM).

4. HIPAA-aligned mitigations across storage, access control, audit logging, and model/LLM
   usage (encryption, role-based access, BAAs for third-party LLMs).

For each factual claim, cite 1-2 authoritative sources. Clearly label any claim not grounded
in the paper or literature as speculative, and note how it could be verified.

Be thorough but focused — a tight analysis, not an exhaustive multi-section treatise.
Do not run code, fetch URLs, or write files. Respond in prose only.

Paper: Wu et al., Nature Medicine 2025, doi:10.1038/s41591-025-03719-2

# Privacy and Data-Security Risk Analysis: PPGR Study (Wu et al., Nature Medicine 2025)

## 1. Sensitive Information Present

The PPGR study (Wu et al., 2025, doi:10.1038/s41591-025-03719-2) collected deeply phenotyped data on 55 participants over multiple weeks, creating a rich dataset with substantial privacy implications:

**Direct identifiers**: The published dataset appears de-identified (participant codes like "P1," "P2"), but the original study necessarily collected names, contact information, and potentially medical record numbers for recruitment, consent, and clinical follow-up. These direct identifiers fall under HIPAA's 18 enumerated identifiers and must be stripped before any data sharing (45 CFR §164.514(b)(2)).

**Quasi-identifiers**: The study reports age (binned into decades in some tables), sex, race/ethnicity, BMI, blood pressure, fasting glucose, HbA1c, and prediabetes/T2D status. Even when binned, combinations of these attributes can uniquely identify individuals, especially in small cohorts. Sweeney (2000, *International Journal of Uncertainty, Fuzziness and Knowledge-Based Systems*) demonstrated that 87% of the U.S. population could be uniquely identified using only ZIP code, birthdate, and sex; the PPGR dataset's metabolic phenotypes are comparably individuating.

**Protected Health Information (PHI)**: Under HIPAA, PHI includes any individually identifiable health information transmitted or maintained in any form. The PPGR dataset contains:
- Continuous glucose monitoring (CGM) time series (minute-by-minute glucose readings over weeks)
- Gut microbiome 16S rRNA abundance profiles (stool samples)
- Clinical labs (lipid panels, liver enzymes, inflammatory markers)
- Dietary intake logs (meal timing, macronutrient composition)
- Wearable-derived heart rate and step counts

All of these constitute PHI when linked to an individual, even indirectly (45 CFR §160.103).

**IRB and data-use restrictions**: The paper states that "all participants provided written informed consent" and that the study was approved by institutional review boards. Typical IRB protocols for deeply phenotyped studies restrict data sharing to:
- Aggregated results only in publications
- Individual-level data shared only under data-use agreements (DUAs) with approved researchers
- Prohibition on re-identification attempts

The Nature Medicine Data Availability statement (standard for this journal) likely requires that individual-level data be deposited in a controlled-access repository (e.g., dbGaP, EGA) or made available "upon reasonable request" with a signed DUA. **Speculative claim**: Without access to the full IRB protocol or DUA, I cannot confirm specific restrictions, but these are standard for human-subjects research involving genomic and metabolic data (NIH Genomic Data Sharing Policy, NOT-OD-14-124).

---

## 2. Elevated Re-identification Risk

Re-identification risk in the PPGR study is substantially higher than in typical clinical datasets due to three compounding factors:

**Small cohort size (n=55)**: Small samples amplify uniqueness. Even after removing direct identifiers, each participant's combination of quasi-identifiers (age, sex, BMI, baseline glucose) may be unique or shared by only 1–2 others. El Emam et al. (2011, *Journal of the American Medical Informatics Association*) showed that re-identification risk increases exponentially as cohort size decreases below 100, particularly when auxiliary information (e.g., public records, social media) is available.

**Highly individuating CGM traces**: Continuous glucose time series are near-unique biometric signatures. Glucose dynamics reflect individual physiology (insulin sensitivity, beta-cell function, gut transit time), diet, activity, and circadian rhythms. Even aggregated features (e.g., mean glucose, variability, time-in-range) can distinguish individuals. **Speculative but grounded**: While I am not aware of a published re-identification attack specifically on CGM da

✅ Tested with Anthropic Claude Sonnet 4.5 (sdo_llm) and verified by our team

Generated by: us.anthropic.claude-sonnet-4-5-20250929-v1:0 (bedrock)
Timestamp: 2026-05-20

What we verified (sound and paper-consistent):
- Correctly identified the three data types (CGM, microbiome, clinical) and the n=55 cohort.
- Demographic breakdown matches the paper's Table 1: sex (27F / 24M / 4 unknown), weight
  status (26 healthy / 18 overweight / 9 obese / 1 underweight), glycemic status
  (27 euglycemic / 26 prediabetes / 1 T2D), ethnicity (37 European / 13 Asian / 4 Hispanic
  / 1 mixed).
- Re-identification reasoning is well grounded: microbiome and dense CGM traces are
  near-unique to individuals (Franzosa 2015; Hall 2018 glucotypes), and small n plus many
  quasi-identifiers raises uniqueness (Sweeney 2000; Rocher 2019). Citations are real.
- Fig 1c is aggregate (per-meal means, not individual traces), so reproducing it is low-risk;
  consistent with HIPAA Expert Determination / k>=10 aggregation principles.
- HIPAA mitigations (encryption, role-based access, audit logging, BAA for any third-party
  LLM, send only aggregate stats) are standard and correct.

Flagged for confirmation (model-generated; NOT yet checked against the paper — do not treat
as fact until verified):
- IRB number (#43883), ClinicalTrials.gov ID (NCT03919877), enrollment dates.
- Device models (Dexcom G4/G6 Pro), sequencing vendor, food-log app, journal license type.
- Exact per-meal CGM curve counts and specific clinical means.

Speculative claims the model self-flagged (correctly): potato-spiker "MASH signature," Asian
rice-spiker genetic explanation, bread-spiker blood-pressure mechanism — all exploratory in
the paper (high pFDR), not established.

## Stage 1 – Clarify the Biological Question

**Goal:** Turn a vague biological curiosity into a concrete, answerable question.

In this CGM carbohydrate-meal study, examples include:

- Do different 50 g-carbohydrate meals produce distinct glycemic responses (AUC above baseline, time-to-peak)?
- Is a person's response to the same meal reproducible across replicate measurements?
- Do clinical markers (insulin resistance, HbA1c) or microbiome composition predict response magnitude?

<details>
<summary style="font-size: 18px; font-weight: 700;">
🧠 Hint: Why clarify the biological question?
</summary>

<div style="
background-color: #eef4ff;
border-left: 6px solid #4a6fa5;
padding: 16px;
border-radius: 8px;
margin-top: 10px;
font-size: 16.5px;
line-height: 1.6;
">

This prompt helps you decide <b>what you want to learn</b> from the proteomics tables <b>before</b> doing any calculations. Clear, testable questions come from matching your biological goal to the <b>labels and identifiers that actually exist</b> in the files (e.g., tissue labels in sample names; protein IDs in the expression matrix and curated list).<br><br>

<b>Practical tip:</b> before proposing questions, quickly verify:
<ul style="margin-top: 8px; margin-bottom: 8px;">
  <li>What are the <b>units</b> on each axis (proteins vs samples)?</li>
  <li>Where do <b>tissue labels</b> come from (often encoded in sample column names)?</li>
  <li>Do identifiers match between files (e.g., curated list uses <b>ProteinID</b> vs gene IDs)? If not, you’ll need a mapping.</li>
</ul>

Avoid questions that require metadata not present (e.g., disease status, genotype, time-course), unless you can confirm those labels exist.

</div>
</details>

🔵 ▶ Run the cell below once and wait (up to 5 minutes) ⏳

In [7]:
%%sdo_llm --show-reply

Study data: continuous glucose monitoring (CGM) curves from 55 participants who each ate
seven standardized 50 g-carbohydrate meals (rice, potatoes, bread, pasta, beans, grapes,
mixed berries) in replicate, alongside paired clinical/metabolic and microbiome profiling.

Task:
- Propose 3-4 precise, answerable questions this dataset can address about postprandial
  glycemic responses (PPGRs) to different carbohydrate meals.
- For each: give the biological rationale and what result would count as a meaningful answer.
- Present as a concise table: Question | Biological Rationale | What Counts as a Meaningful Answer.

Then identify the single question we will reproduce (the basis for Figure 1c) and state it
as one testable hypothesis with explicit variables:
- Outcomes: glucose AUC above baseline, and time from baseline to peak, per meal.
- Exposure: meal type (the seven standardized meals).
- Unit of analysis: per CGM curve (participant x meal x replicate), summarized as
  mean +/- standard error across all curves for each meal.
- Expected direction: starchy meals (rice, potatoes, bread) produce higher AUC and earlier/
  higher peaks than low-glycemic meals (beans, mixed berries), with rice highest overall.

Keep it concise, in prose and one table. Do not run code or fetch URLs.

Paper: Wu et al., Nature Medicine 2025, doi:10.1038/s41591-025-03719-2 (Figure 1c)

Question,Biological Rationale,What Counts as a Meaningful Answer
Q1: Do different 50 g-carbohydrate meals elicit distinct postprandial glycemic response (PPGR) profiles despite matched carbohydrate content?,"Carbohydrate quality (fiber, resistant starch, food matrix) modulates digestion rate and glucose absorption kinetics independently of total carbohydrate load. Glycemic index theory predicts refined starches (rice, bread) cause rapid spikes while intact legumes/fruits produce slower, lower responses.","Statistically significant differences in glucose AUC and/or time-to-peak across meal types (ANOVA/mixed-effects model p < 0.05), with effect sizes >20% between highest and lowest meals, demonstrating that carbohydrate source—not just quantity—determines glycemic impact."
"Q2: Is intra-individual variability in PPGR to the same meal (replicate measurements) smaller than inter-individual variability, and does this differ by meal type?","Personalized nutrition hypotheses propose that individual metabolic/microbiome factors create person-specific responses. If within-person variance << between-person variance, it supports tailored dietary recommendations. Some meals (e.g., beans with fermentable fiber) may show higher inter-individual variance due to microbiome-dependent fermentation.","Variance component analysis showing within-person CV < 15% vs. between-person CV > 30% for most meals, with significantly higher inter-individual variance for beans/berries (p < 0.05 by Levene's test), indicating reproducible personal responses that vary by food type."
"Q3: Do baseline clinical/metabolic markers (fasting glucose, insulin sensitivity, HbA1c) predict the magnitude of PPGR to high-glycemic vs. low-glycemic meals?","Insulin resistance and impaired β-cell function reduce glucose disposal capacity, potentially amplifying responses to rapid glucose loads (high-GI meals) more than slow-release meals. Identifying predictive markers enables risk stratification and targeted meal recommendations.","Significant interaction term (clinical marker × meal type) in mixed-effects regression (p < 0.05), where higher fasting glucose/HbA1c predicts 30–50% greater AUC increase for rice/bread but minimal effect on beans/berries, demonstrating differential metabolic vulnerability by meal type."
"Q4: Does gut microbiome composition (e.g., Prevotella/Bacteroides ratio, fiber-degrading taxa abundance) associate with PPGR magnitude to fiber-rich meals (beans, berries) but not refined starches?","Colonic fermentation of resistant starch and fiber by specific bacterial taxa (Ruminococcaceae, Prevotella) produces short-chain fatty acids that modulate incretin secretion and glucose metabolism. This mechanism should selectively affect responses to fiber-rich foods.","Significant positive correlation (Spearman ρ > 0.4, p < 0.01) between fiber-degrading taxa abundance and bean/berry AUC, but no correlation (ρ < 0.1) with rice/bread AUC, plus mediation analysis showing microbiome explains ≥20% of inter-individual variance in legume/fruit responses."


✅ Tested with Anthropic Claude Sonnet 4.5 (sdo_llm) and verified by our team

Generated by: us.anthropic.claude-sonnet-4-5-20250929-v1:0
Timestamp: 2026-05-21

What we verified:
- The locked hypothesis matches Figure 1c: meal type drives differences in glucose AUC
  above baseline and time-to-peak, summarized as mean +/- SE per meal across all CGM curves.
- Variables correctly specified: outcomes = AUC above baseline + time-to-peak (+ peak
  increment); exposure = the seven standardized 50 g meals; unit = CGM curve
  (participant x meal x replicate).
- Expected direction matches the paper: rice highest overall; starchy meals (rice/potatoes/
  bread) higher than low-GI meals (beans, mixed berries); earlier peaks for starchy meals.
- All four candidate questions are answerable with the study's CGM + clinical + microbiome
  data; Q1 is the one we reproduce for Fig 1c.

Scope note: Q3 (clinical predictors) and Q4 (microbiome associations) are out of scope for
the Fig 1c reproduction but are flagged as possible Stage 5 extensions.

## Stage 2A – Data Pre-Processing

**Goal:** Build a clear mental model of what each table, row, and column represents.

We will work with the data underlying Figure 1c of the CGM carbohydrate-meal study:

- Per-meal postprandial CGM summary data — for each of the seven standardized 50 g meals,
  the glucose-response features extracted from CGM curves (AUC above baseline, time-to-peak,
  peak increment), summarized as mean +/- standard error across all curves.
- Source: the paper's Figure 1 source data (Nature Medicine 2025).

You should be able to answer:

- What does one row and one column represent (per-curve vs per-meal summary)?
- How many meals are represented and how are they labeled?
- How is a raw CGM curve turned into the Fig 1c features (baseline, AUC above baseline, peak, time-to-peak)?
- What does "mean +/- SE across curves" aggregate over (participants x replicates)?

<details>
<summary style="font-size: 18px; font-weight: 700;">
🧠 Hint: Why do we preprocess the data?
</summary>

<div style="
background-color: #eef4ff;
border-left: 6px solid #4a6fa5;
padding: 16px;
border-radius: 8px;
margin-top: 10px;
font-size: 16.5px;
line-height: 1.6;
">

This step is about <b>getting oriented</b>: what each file contains, how it connects to the paper, and what the dataset can (and cannot) answer. It prevents common mistakes like treating processed values as if they were raw measurements or over-interpreting what the data supports.

</div>
</details>


🔵 ▶ Run the cell below once and wait (up to 5 minutes) ⏳

In [8]:
%%sdo_llm --show-reply

Data for reproducing Figure 1c of the CGM carbohydrate-meal study (Wu et al., Nature
Medicine 2025): postprandial glucose-response features for seven standardized 50 g meals
(rice, potatoes, bread, pasta, beans, grapes, mixed berries), derived from CGM curves and
summarized per meal as mean +/- standard error.

Tasks:
- Explain in student-friendly language what this data contains and how a raw CGM curve is
  turned into the Figure 1c features: baseline glucose, AUC above baseline, peak increment,
  and time from baseline to peak.
- State what one row and one column represent, and what the per-meal summary aggregates over
  (participants x meals x replicates -> mean +/- SE per meal).
- Identify three questions this data is well-suited to answer and three it is not, one-line
  reason each.
- Present as a table: Item | Suited? (Yes/No) | Why.

If the Figure 1 source-data file has been uploaded, load it and report its actual structure
(sheet names, columns, number of meals, any missing values); otherwise describe the expected
structure from the paper. Be concise, prose + tables. Do not run code unless the file is
present, and do not fetch URLs.

To do (Stage 4A): structure confirmed by loading fig1.xlsx (sheets 1c_left, 1c_right) in the
reproduction step below.

Paper: Wu et al., Nature Medicine 2025, doi:10.1038/s41591-025-03719-2 (Figure 1c)

Question,Suited?,Why
Which meal produces the highest average peak glucose increment?,Yes,Direct comparison of Peak_increment_mean across meals.
Do beans elicit a lower AUC than bread on average?,Yes,Compare AUC_mean for beans vs bread; SE quantifies uncertainty.
Is time-to-peak longer for pasta than for grapes?,Yes,Compare Time_to_peak_mean; slower-digesting carbs peak later.
Question,Suited?,Why
How does glucose response vary by participant age or BMI?,No,Summary is aggregated over all participants; individual covariates are not included.
"What is the glucose curve shape at 30, 60, 90 minutes post-meal?",No,Only four summary features are reported; full time-series curves are not provided.
Do participants with diabetes respond differently than healthy controls?,No,No stratification by health status in this summary table; would require individual-level data.


✅ Tested with Anthropic Claude Sonnet 4.5 (sdo_llm) and verified by our team

Generated by: us.anthropic.claude-sonnet-4-5-20250929-v1:0
Timestamp: 2026-05-21

Run mode: source-data file not yet uploaded, so the model described the expected structure
from the paper (no data loaded).

What we verified:
- CGM-feature definitions are correct: baseline glucose, AUC above baseline (2 h window,
  positive excursions), peak increment (peak - baseline), and time-to-peak. These are the
  features Fig 1c summarizes.
- The per-meal mean +/- SE framing matches the paper: Fig 1c error bars are 2 standard
  errors and bars are ordered by mean value.
- Per-meal CGM-curve counts confirmed against the paper caption (each dot in Fig 1c):
  rice 115, potatoes 92, bread 99, pasta 65, grapes 98, beans 46, mixed berries 53.
- The suited / not-suited table is accurate: the aggregate table supports cross-meal AUC,
  peak, and time-to-peak comparisons but cannot address per-participant covariates, full
  curve shape, or health-status stratification (those require individual-level data).

To do (Stage 4A): download Source Data Fig. 1 (open access) from the Nature article, upload
to SDO, and confirm sheet/column names so the reproduction loads real values.


## Stage 2B – Integrate & Harmonize

For the GTEx proteome map, QC checks might include:

- Are sample IDs consistent between `protein_expression_mx` and the metadata table?
- How much missingness is there per protein and per tissue?
- Are distributions of intensities comparable across tissues or obviously shifted?
- Are there batch effects related to acquisition batches or processing sites?

### LLM prompt templates (Stage 2B)

1. **QC checklist design**  
   > I have loaded the GTEx proteomics expression matrix from `protein_expression_mx.csv`, along with sample metadata. Design a practical QC checklist for me. Include checks for ID alignment, missing values, distribution anomalies, and potential batch effects across tissues. For each check, explain what might go wrong and how I would detect it.

2. **Summarize QC outcomes**  
   > Here are the results of my QC checks on `protein_expression_mx`:  
   > [PASTE SHORT TEXT OR TABLE SUMMARY]  
   > Summarize the main issues you see, categorize them by severity (critical, moderate, minor), and suggest concrete next steps before proceeding to normalization.


<details>
<summary style="font-size: 18px; font-weight: 700;">
🧠 Hint: Why choose this analysis strategy?
</summary>

<div style="
background-color: #eef4ff;
border-left: 6px solid #4a6fa5;
padding: 16px;
border-radius: 8px;
margin-top: 10px;
font-size: 16.5px;
line-height: 1.6;
">

Before comparing tissues, we need a <b>quality-control checklist</b>. This prompt asks you to plan checks for mismatched IDs, missing values, odd distributions, and batch effects—so you can catch problems early and avoid misleading biological conclusions.

</div>
</details>


🔵 ▶ Run the cell below once and wait (up to 5 minutes) ⏳

In [ ]:
%%sdo_llm --show-reply

Load the GTEx proteomics files:
- `protein_expression_mx.csv` (protein abundance across multiple human tissues)
- `protein_selected_names.csv` (curated list of tissue-enriched proteins)

Task:
- Design a practical QC checklist for this dataset, including checks for:
  - ID alignment between files
  - Missing values
  - Distribution anomalies
  - Potential batch effects across tissues
- For each QC check:
  - Explain what could go wrong (the failure mode).
  - Describe how I would detect the problem using simple summaries, plots, or statistics.
- Present the QC checklist in a clear, structured format (e.g., a table with columns:
  QC Check | What Might Go Wrong | How to Detect It).

<details>
<summary>✅ <strong>Tested by Anthropic Claude Sonnet 4.5 and verified by our team</strong></summary>

Generated by: Anthropic Claude Sonnet 4.5 20250929-v1:0  
Timestamp: 2026-02-26

---

## Protein expression matrix & selected protein names summary (corrected)

### 1) `protein_expression_mx.csv`
- <strong>Shape:</strong> (6108, 210)
- <strong>Description:</strong> <strong>6108 proteins</strong> (rows) × <strong>209 samples</strong> (columns) + a <strong>ProteinID</strong> identifier column.
- <strong>Sample columns:</strong> tissue label is encoded in the column name (e.g., `Adrenal Gland`, `Adrenal Gland.1`, …). Replicates use suffixes like `.1`, `.2`, etc.

### 2) `protein_selected_names.csv`
- <strong>Shape:</strong> (124, 1)
- <strong>Identifier column:</strong> `ProteinID`

### 3) Quick sanity checks
- Missingness in expression matrix is non-trivial (~13%), which is expected in proteomics
- Values are already roughly log-like (range approx -10.5 to 8.05), so treat as log-scale unless you confirm otherwise

</details>

## Stage 3 – Select Data Analysis Strategy (RobNorm-adjacent)

**Goal:** Put tissues and samples onto a comparable scale while preserving meaningful biological differences.

In the full GTEx proteome map analysis, more advanced normalization methods (such as **RobNorm**) were used to handle heterogeneous tissues and technical effects.

In this teaching notebook, we use a **simpler, explainable normalization** as the default:

- log2 transform intensities, and
- median-center samples (and optionally filter proteins with too much missingness).

We treat this as a *conceptual approximation* of the same goals: reduce technical artifacts and make tissues comparable.

### LLM prompt templates (Stage 3)

1. **Compare normalization strategies (including RobNorm)**  
   > Explain, at a high level, three normalization strategies for quantitative proteomics in the GTEx setting: log2 + median centering, quantile normalization, and robust methods like RobNorm. For each, describe what it does, why it might be useful for heterogeneous tissues, and one important limitation.

2. **Normalization diagnostics for GTEx**  
   > Assume I applied log2 + median-centering to `protein_expression_mx`. Describe 3–4 plots or summaries I should examine to verify that normalization behaved as expected, and to check that I did not erase tissue-specific biological signal that the GTEx paper highlights.


<details>
<summary style="font-size: 18px; font-weight: 700;">
🧠 Hint: Why choose this analysis strategy?
</summary>

<div style="
background-color: #eef4ff;
border-left: 6px solid #4a6fa5;
padding: 16px;
border-radius: 8px;
margin-top: 10px;
font-size: 16.5px;
line-height: 1.6;
">

Different tissues can show different baseline shifts and dispersion in protein abundance. This stage helps you choose a strategy (e.g., log2 transform, median-centering, robust offset correction, quantile normalization) and decide which QC plots are appropriate.<br><br>

<b>Key idea:</b> not all normalization methods change the same things:
<ul style="margin-top: 8px; margin-bottom: 8px;">
  <li><b>Offset-only</b> methods mainly correct <b>location</b> (sample baselines / medians).</li>
  <li>Methods that also correct <b>scale</b> can change <b>spread</b> (IQR/MAD/SD) and sometimes the shape of distributions.</li>
</ul>

Your QC checks should match the correction you applied: if you only subtract offsets, expect baseline alignment—not changes in spread.

</div>
</details>

🔵 ▶ Run the cell below once and wait (up to 5 minutes) ⏳

In [ ]:
%%sdo_llm --show-reply

Load the GTEx proteomics files:
- `protein_expression_mx.csv` (protein abundance across multiple human tissues)
- `protein_selected_names.csv` (curated list of tissue-enriched proteins)

Task:
- Explain, at a high level, three normalization strategies for quantitative proteomics in the GTEx setting:
  - log2 + median centering
  - quantile normalization
  - robust methods like RobNorm
- For each strategy, describe:
  - What it does (conceptual overview)
  - Why it might be useful for heterogeneous tissues
  - One important limitation or caveat
- Present the explanations in a concise summary table (e.g.,
  Method | What It Does | Why Useful for Heterogeneous Tissues | Key Limitation).
- Using log2 + median centering as an example, describe 3–4 plots or summary checks
  to verify that normalization behaved as expected and that the tissue-specific
  biological signals emphasized in the GTEx paper are not erased.

<details>
<summary>✅ <strong>Tested by OpenAI GPT-5 and verified by our team</strong></summary>

Generated by: GPT-5  
Timestamp: 2025-12-07 23:25:31  

---

## Expression matrix & protein annotation summary

*(Note: The code block was retried once; both runs produced the same results. Below is the de-duplicated summary.)*

---

### 1. Expression matrix

- **Shape:** (6108, 209)  
- **Inferred index name used for samples:** `Adrenal Gland`  

**First 5 sample IDs:**

- 0.73  
- -0.57  
- 0.91  
- -0.26  
- 1.13  

**First 10 expression columns:**

1. Adrenal Gland.1  
2. Adrenal Gland.2  
3. Adrenal Gland.3  
4. Adrenal Gland.4  
5. Adrenal Gland.5  
6. Adrenal Gland.6  
7. Artery Aorta  
8. Artery Aorta.1  
9. Artery Aorta.2  
10. Artery Aorta.3  

---

### 2. Protein annotation

- **Shape:** (124, 1)  

**First 5 rows:**

| index | x            |
|-------|--------------|
| 0     | ENSG00000036448 |
| 1     | ENSG00000067177 |
| 2     | ENSG00000071794 |
| 3     | ENSG00000073614 |
| 4     | ENSG00000077522 |

**Note:**  
No `protein_col` column was found in the protein annotation file,  
so the column-match check between expression matrix and annotation was skipped.

</details>

## Stage 4A – Data Visualization & Interpretation

**In a nutshell:** Do a chef’s bite, then translate results for everyone at the table.

**What it is:** Stress-test models and translate metrics into domain meaning.

**Why it matters:** Reliable insights require statistical rigor and context-aware interpretation. Pretty plots are not enough; we need to understand uncertainty and practical significance.

**How you do it:** Cross-validate, quantify performance, and interpret effect sizes in plain language.

In a GTEx proteomics setting, this might include:

- Evaluating how stable tissue-enriched protein sets are under resampling or alternative thresholds.
- Checking whether tissue clustering is robust to normalization changes.
- Translating fold-changes and z-scores into meaningful statements about tissue specificity.

### LLM prompt templates (Stage 4A)

1. **Stability of tissue-enriched sets**  
   > I identified a set of tissue-enriched proteins for skeletal muscle using z-score thresholds. Suggest ways to test how stable this set is (for example, resampling donors or perturbing thresholds) and how to visualize the stability.

2. **Explain tissue separation for clinicians**  
   > Here is a PCA plot showing that certain tissues (e.g., liver, brain, muscle) separate clearly in proteome space, while others overlap. Describe these patterns in plain language for a clinical or biological audience, highlighting what this says about tissue specialization.

3. **Effect size interpretation for tissue differences**  
   > Here are a few proteins with their z-scores or log2 fold-changes for a chosen tissue versus others:  
   > [PASTE SMALL TABLE]  
   > Discuss whether these effect sizes are likely to be biologically meaningful, relating them to known tissue functions when possible, and mention at least two caveats.


<details>
<summary style="font-size: 18px; font-weight: 700;">
🧠 Hint: What do these visualizations help us see?
</summary>

<div style="
background-color: #eef4ff;
border-left: 6px solid #4a6fa5;
padding: 16px;
border-radius: 8px;
margin-top: 10px;
font-size: 16.5px;
line-height: 1.6;
">

These visualizations help you test whether your “tissue-enriched protein” results are <b>stable and repeatable</b>, rather than an artifact of one cutoff or a few samples. A typical workflow is:
<ul style="margin-top: 8px; margin-bottom: 8px;">
  <li>Find candidate tissue-enriched proteins (e.g., using a simple <b>z-score</b> rule).</li>
  <li>Check <b>sensitivity</b> to the threshold (does the set change drastically if z changes from 1.75 to 2.0?).</li>
  <li>Check <b>robustness</b> via bootstrap/subsampling (do the top proteins keep reappearing?).</li>
</ul>

<b>Important:</b> If your normalization is <b>offset-only</b> (subtracting a per-sample constant), it will mostly change <b>sample medians / baseline shifts</b>, but it should <b>not</b> change spread metrics like <b>IQR/MAD</b>. If a “spread plot” changes a lot, double-check you didn’t compute the metric on the wrong axis or unintentionally apply a scaling step.

</div>
</details>

🔵 ▶ Run the cell below once and wait (up to 5 minutes) ⏳

In [ ]:
%%sdo_llm --show-reply

Reproduce Figure 1c from the CGM carbohydrate-meal study (Wu et al., Nature Medicine 2025)
using the file fig1.xlsx in the working directory.

DATA:
- Sheet "1c_left":  columns foods | feature | mean | se*2  (feature = AUC_above_baseline)
- Sheet "1c_right": columns foods | feature | mean | se*2  (feature = time_to_peak)
- Each sheet has 7 rows, one per meal: Rice, Potatoes, Bread, Pasta, Grapes, Beans, Berries.
- "se*2" is already 2 x standard error (the paper's error-bar definition) - plot it directly.

VALIDATE FIRST (print): for each sheet, its shape, the 7 food names, and confirm mean/se*2
are numeric with no missing values. STOP and explain if a sheet lacks exactly 7 meals.

PLOT (two panels side by side, matching Fig 1c):
- Left: bar chart of AUC_above_baseline mean per meal, error bars = se*2,
  y-axis "Glucose AUC above baseline", bars ordered by descending mean.
- Right: bar chart of time_to_peak mean per meal, error bars = se*2,
  y-axis "Time from baseline to peak (min)", bars ordered by descending mean.
- Label each bar with its meal name. Inline plot only; do not save files.

REPORT:
- The meal ranking by AUC and by time-to-peak.
- Whether rice has the highest AUC (the paper's headline result).
- One scientific interpretation sentence plus one plain-language sentence per panel.

Be concise. Do not fetch URLs.

Paper: Wu et al., Nature Medicine 2025, doi:10.1038/s41591-025-03719-2 (Figure 1c)

<details>
<summary>✅ <strong>Tested with Anthropic Claude Sonnet 4.5 and verified by our team</strong></summary>

Generated by: us.anthropic.claude-sonnet-4-5-20250929-v1:0 (bedrock)  
Timestamp: 2026-02-26 21:45:39  

This verified run checks whether an offset-only, RobNorm-like robust correction reduces systematic baseline shifts across tissues without collapsing biological variation.

### Data validation (required checks)

- **Expression matrix shape:** (6108, 211)  
  - First 10 columns: `ProteinID`, `Adrenal Gland`, `Adrenal Gland.1`, …, `Artery Aorta`, `Artery Aorta.1`  
  - First 5 ProteinIDs: `ENSG00000000419`, `ENSG00000000971`, `ENSG00000001036`, `ENSG00000001084`, `ENSG00000001497`
- **Samples:** 210 (all columns excluding `ProteinID`)
- **Tissues parsed from sample names:** 32 unique tissues  
  - Top counts: Muscle Skeletal (13), Colon Sigmoid (12), Stomach (12), Artery Aorta (11), Heart Atrial (11), Thyroid (11)  
  - Minimum count: 1 (Minor Salivary)
- **PASS:** Muscle Skeletal has 13 samples (≥10)
- **Missingness:** 13.14% NaNs (acceptable for proteomics; all summary statistics are NaN-aware)

### Scale determination

Global quantiles of `X_raw`:
- 1%: -4.55  
- 50%: -0.16  
- 99%:  2.82  

**Conclusion:** data is already **log-scale** (centered near 0, moderate range).  
**Scale used:** log-scale (as-is)

### Pre-normalization statistics (per-sample, across proteins)

- `raw_median`: min=-1.810, median=-0.165, max=0.500  
- `raw_iqr`:    min=0.660, median=1.260, max=2.365  

### Robust offset correction (simplified RobNorm-like)

- `offset`: min=-1.502, median=-0.021, max=0.567  
- Output matrix: `X_norm` shape (6108, 210)

### Post-normalization statistics (per-sample spread)

- `norm_iqr`: min=0.660, median=1.260, max=2.365  

**Note (important):** This equality is **expected** for offset-only normalization: subtracting a constant per sample changes the **center** (medians/offsets) but not **spread** metrics like IQR/MAD/SD.

---

## Verified plots (Figure 4A panels)

> Replace the `src` paths below with the correct local/relative paths where you copy the attached figures.

### Plot 1 — Per-tissue distribution of raw sample medians (pre-normalization)

**Scientific interpretation:** Per-tissue variation in raw sample medians indicates systematic baseline shifts across tissues, likely reflecting technical effects (extraction efficiency, tissue composition, batch effects).  
**Lay explanation:** Different tissues start with different “average” protein levels, suggesting technical shifts that should be corrected.


<img src="materials/images/Figure1-4A.png" alt="Plot 1: Per-tissue distribution of raw sample medians (pre-normalization)" style="max-width: 100%; border: 1px solid #ddd; border-radius: 8px; margin-top: 10px;"/>

---

### Plot 2 — Per-tissue distribution of normalization offsets

**Scientific interpretation:** The offsets quantify how much baseline correction is applied per sample; tissue patterns reveal which tissue groups deviated most from the robust reference.  
**Lay explanation:** This shows how much adjustment was needed in each tissue; bigger offsets mean those samples were further from the “typical” baseline.

<img src="materials/images/Figure2-4A.png" alt="Plot 2: Per-tissue distribution of normalization offsets" style="max-width: 100%; border: 1px solid #ddd; border-radius: 8px; margin-top: 10px;"/>

---

### Plot 3 — Raw vs normalized IQR distributions (selected tissues)

Selected tissues: Muscle Skeletal, Colon Sigmoid, Stomach, Artery Aorta

**Scientific interpretation (updated for offset-only correction):** Overlap between raw and normalized IQR distributions is expected because offset-only correction shifts sample baselines without changing within-sample spread.  
**Lay explanation:** The “spread” of protein values stays the same because we only shifted each sample up/down, not stretched or compressed it.

<img src="materials/images/Figure3-4A.png" alt="Plot 3: Raw vs normalized IQR distributions (selected tissues)" style="max-width: 100%; border: 1px solid #ddd; border-radius: 8px; margin-top: 10px;"/>

---

## Final quantitative check (key QC criterion)

Variance of per-tissue medians:
- **Before correction:** 0.164601  
- **After correction:**  0.003882  
- **Reduction:** 97.64%

**PASS:** Baseline differences across tissues were strongly reduced but not eliminated—consistent with a robust correction that removes systematic shifts while retaining biological structure.

---

## Notes

- Many tissues have <10 samples (e.g., Minor Salivary=1). This is acceptable for GTEx-style datasets, but downstream analyses (e.g., classifiers) should either (a) exclude rare tissues or (b) aggregate them.
- Matplotlib warning in the run is non-substantive (renamed `labels` parameter); it does not affect the interpretation.

</details>


## Stage 4B – Communicate Findings

**In a nutshell:** Plate the meal so it looks appetizing and the ingredients are clear.

**What it is:** Convert statistics and model outputs into concise, trustworthy stories.

**Why it matters:** Good visuals and clear writing accelerate understanding and drive action. Ambiguous or overloaded figures create confusion and mistrust.

**How you do it:** Build a minimal figure set, quantify uncertainty, and craft a plain-language abstract that respects limitations and avoids unwarranted clinical claims.

In the GTEx proteome map context, this might mean:

- Showing a small number of plots that clearly illustrate tissue clustering and tissue-enriched proteins.
- Highlighting a few exemplar proteins and explaining their tissue roles.
- Explicitly connecting your simplified analysis to the more comprehensive results of the GTEx paper.

### LLM prompt templates (Stage 4B)

1. **Design a GTEx-style figure set**  
   > Based on my GTEx proteomics analysis, I want at most three plots for a non-technical audience. Design a figure set (≤3 plots) that communicates the main findings about tissue clustering and tissue-enriched proteins. For each plot, specify what goes on each axis, how points or tissues are colored, what message the viewer should take away, and one potential misinterpretation to guard against.

2. **Draft a short abstract**  
   > Using the key results from this GTEx proteomics analysis: [PASTE BULLET POINTS], draft a 150-word abstract summarizing the main findings and their implications. Write in plain language, avoid causal claims, and explicitly mention at least two limitations of this simplified analysis relative to the original GTEx proteome map paper.

3. **Contrast baseline vs tissue-specific deviation**  
   > I want a visualization that contrasts a "baseline" proteomic profile (for example, the median across all tissues) with a "deviation" profile for a particular tissue (for example, liver or brain). Propose one or two visualization designs (e.g., heatmaps, radar plots, or difference plots) and explain what patterns they would make easy to see.


<details>
<summary style="font-size: 18px; font-weight: 700;">
🧠 Hint: How to interpret and communicate the key findings
</summary>

<div style="
background-color: #eef4ff;
border-left: 6px solid #4a6fa5;
padding: 16px;
border-radius: 8px;
margin-top: 10px;
font-size: 16.5px;
line-height: 1.6;
">

This step focuses on <b>communication</b>. You’ll design a few audience-friendly plots, draft a short plain-language abstract, and note limitations—so the takeaways are understandable without over-claiming (especially avoiding causal language).<br>

<b>Before writing the abstract:</b> make sure your upstream checks passed (e.g., tissue labels were parsed correctly and the enrichment set is stable across thresholds / resampling). If those checks fail, treat the “findings” as preliminary and revise the analysis first.

</div>
</details>

🔵 ▶ Run the cell below once and wait (up to 5 minutes) ⏳

In [ ]:
%%sdo_llm --show-reply

Load the GTEx proteomics files:
- `protein_expression_mx.csv` (protein abundance across multiple human tissues)
- `protein_selected_names.csv` (curated list of tissue-enriched proteins)
- `protein_expression_mx.csv` is shaped (n_proteins, n_samples).
- The first column is `ProteinID` (protein identifiers). Treat it as the row key / index.
- All other columns are sample columns; the tissue label is encoded in the column name
 (e.g., "Muscle Skeletal", "Muscle Skeletal.1", ...).

Task:
- Before any analysis, VERIFY the matrix orientation and STOP if it is not as expected.
- Define skeletal muscle samples as columns whose name starts with EXACTLY:
 "Muscle Skeletal" (do NOT include "Esophagus Muscle" unless explicitly asked).
- Identify a set of tissue-enriched proteins for skeletal muscle using a z-score–based
  thresholding approach (e.g., high z-score in muscle relative to other tissues).
- Propose simple ways to test how stable this tissue-enriched set is, such as resampling
  or subsampling donors, perturbing the z-score threshold, or comparing results across bootstrap runs
- Create and display inline (do not save any files or call functions like plt.savefig):
  - One plot showing the top skeletal muscle–enriched proteins (e.g., bar plot of z-scores).
  - One plot visualizing stability (e.g., how often each protein is selected across resamples
    or thresholds).
- Accompany each plot with a brief 1–2 sentence explanation describing
  what it shows and how to interpret it, both as a scientific summary and in lay terms.

<details>
<summary>✅ <strong>Tested with Anthropic Claude Sonnet 4.5 and verified by our team</strong></summary>

Generated by: us.anthropic.claude-sonnet-4-5-20250929-v1:0 (bedrock)  
Timestamp: 2026-02-26 21:47:22  

This verified run identifies skeletal muscle–enriched proteins using a simple z-score rule (muscle vs all other tissues) and evaluates robustness using bootstrap resampling, subsampling, and threshold sensitivity.

### 1) Matrix orientation + tissue parsing (sanity checks)

- **Expression matrix shape:** (6108, 211)  
- **First column:** `ProteinID` (row identifiers) ✅  
- **Sample columns:** 210 ✅  
- **Skeletal muscle samples:** 13 ✅  
  - Examples: `Muscle Skeletal`, `Muscle Skeletal.1`, `Muscle Skeletal.2`  
- **Non-muscle samples:** 197 ✅  

Top tissues by sample count:
- Muscle Skeletal (13), Colon Sigmoid (12), Stomach (12), Artery Aorta (11), Heart Atrial (11), Thyroid (11), GE junction (10), Pancreas (9), Skin Unexpo (9), Esophagus Muscle (8)

**Conclusion:** Matrix orientation and tissue parsing are consistent with the intended format (proteins × samples; tissue encoded in column names).

---

## 2) Muscle-enriched protein detection (z-score > 2.0)

- **Threshold:** z-score > 2.0  
- **Number of muscle-enriched proteins:** 55  

Top muscle-enriched proteins (by z-score; top 20 shown in console output):
- ENSG00000073614 (z = 5.82)
- ENSG00000183091 (z = 2.84)
- ENSG00000143549 (z = 2.75)
- ENSG00000086967 (z = 2.70)
- ENSG00000149925 (z = 2.65)
- ENSG00000196465 (z = 2.63)
- ENSG00000196091 (z = 2.62)
- ENSG00000120729 (z = 2.60)
- ENSG00000239474 (z = 2.49)
- ENSG00000196218 (z = 2.49)
- … (see run log for full list)

---

## 3) Stability and sensitivity analysis (robustness checks)

### Bootstrap resampling (n=100)
- Selected in >90% of bootstraps: 28 proteins  
- Selected in >50% of bootstraps: 54 proteins  
- Selected in <10% of bootstraps: 59 proteins  

### Threshold sensitivity (number of enriched proteins)
- z > 1.5: 104  
- z > 1.75: 83  
- z > 2.0: 55  (selected threshold)  
- z > 2.25: 24  
- z > 2.5: 8  
- z > 3.0: 1  

### Core set across thresholds
- Core proteins (present at z > 1.75, 2.0, and 2.25): 24 proteins

### Subsampling (80% of donors, n=50)
- Selected in >90% of subsamples: 40 proteins  
- Selected in >50% of subsamples: 53 proteins  

### Agreement between resampling strategies
- Bootstrap vs subsample correlation: 0.774  
- Interpretation: high agreement indicates the enrichment signal is not dependent on one specific resampling method.

---

## Verified plots (Figure 4B panels)

> Replace the `src` paths below with the correct local/relative paths where you copy the attached figures.

### Plot 1 — Top 25 skeletal muscle–enriched proteins (z-score > 2.0)

**Scientific interpretation:** This bar plot shows the proteins with the highest z-scores in skeletal muscle relative to all other tissues. A higher z-score means the protein is more strongly elevated in muscle compared to the global distribution across tissues.  
**Lay explanation:** These are proteins that are most strongly “turned on” in skeletal muscle compared with other tissues—useful as a molecular signature of muscle cells.

<img src="materials/images/TOP25_MUSCLE_ENRICHED-4B.png" alt="Top 25 skeletal muscle–enriched proteins (z-score > 2.0)" style="max-width: 100%; border: 1px solid #ddd; border-radius: 8px; margin-top: 10px;"/>

---

### Plot 2 — Stability and sensitivity diagnostics (4-panel figure)

Panel meanings:
- **A:** Bootstrap selection frequency for the top proteins (how often each protein reappears)
- **B:** Number of enriched proteins as the z-threshold changes (sensitivity)
- **C:** Distribution of bootstrap stability frequencies (how many are stable vs unstable)
- **D:** Bootstrap vs subsampling agreement (consistency across resampling strategies)

**Scientific interpretation:** The combined stability diagnostics show that a subset of muscle-enriched proteins is highly reproducible (>90% bootstrap frequency), while many lower-ranked candidates are threshold-sensitive and unstable.  
**Lay explanation:** We repeatedly re-run the analysis in slightly different ways to see if we get the same muscle proteins. Proteins that keep showing up are likely real muscle markers; ones that come and go are less reliable.

<img src="materials/images/STABILITY_PANEL-4B.png" alt="Stability and sensitivity analysis (bootstrap, thresholding, subsampling)" style="max-width: 100%; border: 1px solid #ddd; border-radius: 8px; margin-top: 10px;"/>

---

## 4) Verified summary of findings (what to use downstream)

- **Highly stable muscle-enriched proteins:** 28  
  - These appear in >90% of bootstraps and are best for downstream analyses.
- **Moderately stable proteins:** 26  
  - Appear in 50–90% of resamples; useful for exploratory work.
- **Unstable proteins:** 92  
  - Appear in <50% of resamples; likely false positives or threshold-sensitive.

**Recommendation:** Use the >90% bootstrap set as the “high-confidence” muscle signature; keep the 50–90% set for hypothesis generation. The chosen threshold (z > 2.0) is supported by the sensitivity curve and retains a manageable number of candidates (55).

---

## IMPORTANT NOTE ABOUT FOLD-CHANGE REPORTING

The console log printed negative “FC” values for some proteins. A fold-change expressed as a ratio should never be negative.

If you want to report effect sizes, use one of the following:
- **log2FC (signed):** mean(muscle) − mean(non-muscle) on the log2 scale (can be negative)  
- **FC (ratio):** 2^(log2FC) (always positive)

This does not affect the z-score ranking or the stability diagnostics, but it should be corrected before presenting fold-change values in any write-up.

</details>

## Stage 5 – Pushing Boundaries and New Insights

**In a nutshell:** Collect feedback and refine the recipe. Re-run ethics when scope changes.

**What it is:** A structured feedback loop for refining data, models, and hypotheses. Any expansion in scope (new variables, new outcomes, external data) should trigger a return to Stage 0.

**Why it matters:** Data drift, evolving cohorts, and new measurements are constant. Iteration preserves accuracy, fairness, and relevance over time.

**How you do it:** Solicit feedback from domain experts, audit bias, add or refine signals, and loop back through earlier stages as needed.

### LLM prompt templates (Stage 5)

1. **Peer-review style critique**  
   > Imagine you are a peer reviewer assessing this GTEx-based proteomics study. List three follow-up analyses or sensitivity checks you would request to strengthen the conclusions. For each request, briefly explain what risk or uncertainty it addresses.

2. **Signal extension to multi-omics**  
   > Suggest how integrating additional data (for example, matching transcriptomics, metabolomics, or key clinical variables) could improve the robustness or interpretability of my proteomics findings from GTEx. Propose two concrete extension projects, each with a short sequence of steps.

3. **Fairness and subgroup performance**  
   > Assume I built a predictive model using GTEx proteomics features (for example, classifying tissue type). Propose a fairness or robustness audit that compares performance across relevant subgroups (for example, tissue classes with fewer samples or donors). Describe what metrics to look at, how to summarize differences, and how to decide whether they are concerning.


<details>
<summary style="font-size: 18px; font-weight: 700;">
🧠 Hint: Why extend beyond reproduction?
</summary>

<div style="
background-color: #eef4ff;
border-left: 6px solid #4a6fa5;
padding: 16px;
border-radius: 8px;
margin-top: 10px;
font-size: 16.5px;
line-height: 1.6;
">

After reproducing a small slice of the paper, this prompt asks you to think like a <b>peer reviewer</b>: propose follow-up analyses or sensitivity checks that would increase confidence and clarify what might change under different assumptions.

</div>
</details>


🔵 ▶ Run the cell below once and wait (up to 5 minutes) ⏳

In [ ]:
%%sdo_llm --show-reply

- Imagine you are a peer reviewer assessing this GTEx-based proteomics study.
  List three follow-up analyses or sensitivity checks you would request to strengthen the conclusions,
  and briefly explain what risk or uncertainty each one addresses.

<details>
<summary>✅ <strong>Tested by OpenAI GPT-5 and verified by our team</strong></summary>

Generated by: GPT-5  
Timestamp: 2025-12-07 23:37:14  

---

## How added data helps (high level)

- **Matching transcriptomics (RNA-seq)**  
  Checks whether protein changes reflect transcriptional regulation (concordant) or post-transcriptional control (discordant), increases confidence in protein-level hits, and improves pathway-level signal by aggregating across omics.

- **Metabolomics / metabolite–gene links**  
  Anchors proteins to biochemical function and pathways, helps generate mechanistic hypotheses (enzyme → substrate), and prioritizes proteins that map to altered metabolites.

- **Sample/donor clinical variables & metadata**  
  Control for confounders (age, sex, tissue, batch), enable stratified/replication analyses, and increase translational relevance (biomarker subgroups).

- **Gene set / pathway annotations (GMT files)**  
  Move from individual proteins to coherent pathway-level changes, reducing multiple-testing noise and improving interpretability.

---

## Two concrete extension projects (concise summary table)

| Project Name                                      | Data Types Integrated                                                                                                                                                                                                                          | Key Steps                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | How It Improves Interpretability / Robustness                                                                                                                                                                                                                                  |
|---------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Proteogenomic concordance & post-transcriptional regulation | - Proteomics (`protein_expression_mx.csv`) + `protein_selected_names.csv` (map to gene symbols)  <br> - RNA-seq counts (`rsem.merged.transcript_counts.tsv`)  <br> - GENCODE mapping (`gencode.v45.metadata.HGNC`)  <br> - Sample metadata (`GSE158739_metadata.csv`) | 1) Map protein columns → `gene_symbol` using `protein_selected_names.csv`.  <br> 2) Load RSEM transcript counts, collapse transcripts → gene-level expression via GENCODE mapping (sum or tximport-like approach).  <br> 3) Normalize RNA (e.g., TMM/TPM + log1p) and proteomics (z-score or quantile normalize), perform QC & remove low-quality samples; align samples using GSE metadata (match `SampleID` → RNA sample IDs).  <br> 4) Filter to genes/proteins measured in both layers; compute per-sample and per-gene RNA–protein correlation (Spearman/Pearson).  <br> 5) Identify: (a) concordant hits (both RNA↑ & protein↑), (b) discordant candidates (RNA↑ but protein↘ or vice versa).  <br> 6) Perform pathway-level integration (aggregate gene-wise signals into pathways using GMTs), and optionally run multi-omics factor analysis (MOFA/PLS-DA/DIABLO) to extract joint latent factors.  <br> 7) Validate top discordant proteins with prior knowledge (e.g., protein stability, PTMs). | - Distinguishes transcriptional vs post-transcriptional regulation (biologically informative).  <br> - Increases robustness: proteins supported by consistent RNA signal are more likely true positives; discordant cases reveal regulation not captured by RNA alone.  <br> - Pathway-level concordance improves signal-to-noise and reduces false discoveries. |
| Metabolic anchoring + clinical stratification     | - Proteomics (`protein_expression_mx.csv`) + `protein_selected_names.csv`  <br> - Metabolite–gene association table (`Tab2_42255_2023_817_MOESM3_ESM.csv`)  <br> - Pathway gene sets (c2/h.all GMTs)  <br> - Sample/donor metadata (`GSE158739_metadata.csv`) and any available clinical covariates | 1) Annotate proteins with gene symbols and join to Tab2 metabolite–gene associations (use `metabolite` ↔ gene links).  <br> 2) For each protein, compile known metabolite links and per-tissue effect columns from Tab2 to prioritize metabolite-associated proteins.  <br> 3) Perform protein-level enrichment of metabolite-linked gene sets (use GMTs to move to pathways).  <br> 4) Use metadata to stratify analyses by tissue, sex, or clinical covariates and test if metabolite-linked protein signatures are consistent across strata.  <br> 5) Build protein↔metabolite inferred networks (edge = known association from Tab2 or high cross-sample correlation where metabolomics available).  <br> 6) Run mediation or regression models adjusting for clinical covariates to assess whether protein changes mediate associations between clinical variables and pathway activity.  <br> 7) Prioritize candidate mechanistic proteins for follow-up (e.g., enzymes with both proteomic change and metabolite association). | - Anchors proteomic hits to biochemistry (metabolite links give mechanistic context).  <br> - Clinical stratification shows which associations generalize vs are confounded, improving reproducibility.  <br> - Network and mediation analyses identify likely causal/functional players rather than mere markers. |


</details>

---
Copyright (c) 2026 Stanford Data Ocean (SDO)

All rights reserved.